transaction/core/src/encrypted_fog_hint.rs

```diff
@@ -103,12 +103,31 @@ impl EncryptedFogHint {
     /// random plaintext. There are several ways we could sample that distribution
     /// but the simplest is to do exactly that. This is also future-proof if we later
     /// tweak the cryptobox implementation.
-    pub fn fake_onetime_hint<T: RngCore + CryptoRng>(rng: &mut T) -> Self {
+    /// Optionally provide the random plaintext buffer to be encrypted.
+    pub fn fake_onetime_hint<T: RngCore + CryptoRng>(
+        rng: &mut T,
+        plaintext_buf: Option<&[u8]>,
+    ) -> Self {
         // Make plaintext of the right size
-        let plaintext = GenericArray::<
-            u8,
-            Diff<EncryptedFogHintSize, <VersionedCryptoBox as CryptoBox<Ristretto>>::FooterSize>,
-        >::default();
+        let plaintext = if let Some(pt) = plaintext_buf {
+            // Truncate the plaintext to a buffer of the correct size
+            GenericArray::<
+                u8,
+                Diff<
+                    EncryptedFogHintSize,
+                    <VersionedCryptoBox as CryptoBox<Ristretto>>::FooterSize,
+                >,
+            >::clone_from_slice(pt)
```

Comment:
this will panic if plaintext_buf is not exactly the right size

Reply:
I ran into that in testing as well - the len is 38 bytes, which was a bit hard to track down, as it's referencing [Ristretto(u32) + TagSize](https://github.com/mobilecoinfoundation/mobilecoin/blob/master/crypto/box/src/hkdf_blake2b_aes_128_gcm.rs#L21). Will see if I can derive the expected len and then slice to the correct len before sending to GenericArray. Otherwise, I can note in a comment that the new fn will panic if not passed the correct size.